File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
# 프로그래머스 12909 올바른 괄호 LV.2
def solution(s):
    answer = True
    stack = []
    for c in s:
        if c == '(':
            stack.append(c)
        else:
            if len(stack) == 0:
                return False
            else:
                stack.pop()

    return len(stack) == 0

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        
        char[] c = s.toCharArray();
        for (int i = 0; i < c.length; i++) {
            if (c[i] == ')') {
                if (stack.isEmpty()) return false;
                if (stack.peekLast() == '(') stack.removeLast();
            } else stack.addLast(c[i]);
        }
        
        return stack.isEmpty();
    }
}